superset/db_engine_specs/mssql.py

```diff
@@ -50,8 +50,10 @@ def epoch_to_dttm(cls):
         return "dateadd(S, {col}, '1970-01-01')"
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
-        return "CONVERT(DATETIME, '{}', 126)".format(dttm.isoformat())
```

Comment:
We shouldn't be casting all types to a  `DATETIME`. The reason for this function is to ensure that the LHS and RHS for the time filter comparison are equivalent types.